packages/db/src/controller/level.ts

```diff
@@ -175,6 +190,26 @@ export class LevelDbController implements IDatabaseController<Uint8Array, Uint8A
 
     this.metrics?.dbWriteItems.inc({bucket}, itemsRead);
   }
+
+  /** Start interval to capture metric for db size */
+  private collectDbSizeMetric(): void {
+    this.dbSizeMetric();
+    this.dbSizeMetricInterval = setInterval(this.dbSizeMetric.bind(this), DB_SIZE_METRIC_INTERVAL_MS);
+  }
+
+  /** Capture metric for db size */
+  private dbSizeMetric(): void {
+    const minKey = Buffer.from([0x00]);
+    const maxKey = Buffer.from([0xff]);
+
+    this.approximateSize(minKey, maxKey)
```

Comment:
I did some tests back in the day and this was both extremelt expensive and innacurate. Please test in depth before consider merging

Reply:
> I think it makes sense to test this on a mainnet archive node, is there any way we could do this?

Yes sure, provision a node and sync from genesis.

Reply:
Did some testing on a mainnet node with 30GB db, `approximateSize` still executes mostly under 1ms

```
mainnet-consensus-1  | approximateSize: 0.182ms
mainnet-consensus-1  | approximateSize: 0.596ms
mainnet-consensus-1  | approximateSize: 0.592ms
mainnet-consensus-1  | approximateSize: 0.211ms
mainnet-consensus-1  | approximateSize: 0.923ms
mainnet-consensus-1  | approximateSize: 1.36ms
mainnet-consensus-1  | approximateSize: 1.185ms
mainnet-consensus-1  | approximateSize: 0.961ms
mainnet-consensus-1  | approximateSize: 0.213ms
```